### ISSUE TITLE
Non-readonly metadata is shared between endpoints

### ISSUE THREAD

**Issue by Kahbazi** [microsoft/reverse-proxy#1778]
### Describe the bug
When creating the endpoints the default metadata for authorization and cors is created once to reuse for multiple endpoints in order to save object allocation, but I just noticed that `IAuthorizeData` and `IEnableCorsAttribute` have setter in their properties. So one might change this metadata in order to change it for the specific endpoint but it will affect every other endpoints as well. 

https://github.com/microsoft/reverse-proxy/blob/b9f43dd4b383e7c2e1b222c0dd6587a848d88a56/src/ReverseProxy/Routing/ProxyEndpointFactory.cs#L22-L23

### To Reproduce
I haven't seen any framework that actually change theses metadata during runtime, but the code might be something like this:

```csharp
var dataSources = serviceProvider.GetServices<EndpointDataSource>();
var endpoint = dataSources.SelectMany(x => x.Endpoints).Where(x => ...).First();
endpoint.Metadata.GetMetadata<IAuthorizeData>().AuthenticationSchemes = "SomethingNew";
```

**Comment by karelz** [comment]
Triage: Looks like theoretical problem to us. Metadata should not be modified on endpoints.
Do you have a legitimate reason to change Metadata like this @Kahbazi?

**Comment by Kahbazi** [comment]
> Triage: Looks like theoretical problem to us. Metadata should not be modified on endpoints.

I agree. The ideal solution would be to remove the setter from the metadata in ASP.Net Core repo. Though it would be a breaking change.

> Do you have a legitimate reason to change Metadata like this @Kahbazi?

I'm not doing this myself, maybe someone wants to change policy name during runtime but it could be done other ways beside changing the metadata.
I just reported this to prevent someone hitting this bug in future.
